_posts/2018-02-06-unrealiable-network.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title: Quick Tips for App Developers on Surviving with Unrealiable Network
+author: jarzka
+excerpt: Methods that you, application developer, can do to make your app feel more stable even if the underlying network is unreliable.
+tags:
+- clojure
+- clojurescript
+- network programming
+---
+
+You cannot always choose where your application will be hosted. Sometimes the network infrastructure can be unrealiable, meaning that network requests can sometimes take a long time to complete and requests will fail at random rates. Even if these problems could not be prevented from happening, luckily we, application developers, have some tricks we can use to make our applications *feel* more stable in unrealiable network. And even if your network infrastucture is solid, it is not guaranteed that occasional communication failures won't happen, as illustrated in [Two Generals Problem](https://en.wikipedia.org/wiki/Two_Generals%27_Problem). Thus, these quick tips, targeted mainly for single page web application and mobile application developers, will probably help your application behave more stable and quickly for your end users.
+
+# Avoid Making Network Requests as Much as You Can
+
+The best way to avoid getting failing network requests is of course to abandom making them completely.
+
+![Failing network requests meme](/img/unrealiable-network/network_requests_meme.jpg)
+
+Thought this is very efficient, it is not very practical for most of us. Majority of web and mobile apps depend on getting data from servers, and thus, making network requests is mandatory. Still, if we cannot abandom them completely, we can at least reduce the amount of them or optimise them to achieve a good compromise.
```

Comment:
> Thought this is

Even though this is

> Still, if we cannot abandom them completely, we can at least reduce the amount of them or optimise them to achieve a good compromise.

Still, even if we cannot avoid them completely, it is possible to reduce the number of or otherwise optimise the requests in order to achieve a suitable compromise